src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -0,0 +1,311 @@
+﻿using OpenTelemetry;
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.OpenTelemetry;
+
+// https://develop.sentry.dev/sdk/performance/opentelemetry
+
+/// <summary>
+/// Sentry span processor for Open Telemetry.
+/// </summary>
+public class SentrySpanProcessor : BaseProcessor<Activity>
+{
+    private readonly IHub _hub;
+
+    // ReSharper disable once MemberCanBePrivate.Global - Used by tests
+    internal readonly ConcurrentDictionary<ActivitySpanId, ISpan> _map = new();
+    private readonly SentryOptions? _options;
+    private readonly Lazy<IDictionary<string, object>> _resourceAttributes;
+
+    /// <summary>
+    /// Constructs a <see cref="SentrySpanProcessor"/>.
+    /// </summary>
+    public SentrySpanProcessor(IHub hub)
+    {
+        _hub = hub;
+        _options = hub.GetSentryOptions();
+
+        if (_options?.Instrumenter != Instrumenter.OpenTelemetry)
+        {
+            throw new InvalidOperationException(
+                $"To use the {nameof(SentrySpanProcessor)}, you must also set the " +
+                $"{nameof(SentryOptions.Instrumenter)} option to {nameof(Instrumenter.OpenTelemetry)} " +
+                "when initializing Sentry.");
+        }
+
+        // Resource attributes are consistent between spans, but not available during construction.
+        // Thus, get a single instance lazily.
+        _resourceAttributes = new Lazy<IDictionary<string, object>>(() =>
+            ParentProvider?.GetResource().Attributes.ToDictionary() ?? new Dictionary<string, object>(0));
+
+        hub.ConfigureScope(scope => scope.AddTransactionProcessor(TransactionProcessor));
```

Comment:
So they way I've wired this up is:
- Added a new internal `SentryOptions.ImplicitTransactionProcessors` property
- Created a new [OpenTelemetryTransactionProcessor](https://github.com/getsentry/sentry-dotnet/blob/feat/open-telemetry/src/Sentry.OpenTelemetry/OpenTelemetryTransactionProcessor.cs)
- Modified `SentryOptions.Instrumenter`  to be `{ get; internal set; }`
- Added the following extension method that people would use instead of setting that Instrumenter property directly:

```csharp
public static class SentryOptionsExtensions
{
    /// <summary>
    /// Enables OpenTelemetry instrumentation with Sentry
    /// </summary>
    /// <param name="options"></param>
    public static void UseOpenTelemetry(this SentryOptions options)
    {
        options.Instrumenter = Instrumenter.OpenTelemetry;
        options.ImplicitTransactionProcessors.Add(
            new OpenTelemetryTransactionProcessor()
            );
    }
}
```

I'm not sure how we feel about this as it's a small departure from how this gets wired up in other Sentry SDKs. Open to other suggestions if we don't like this though.

Reply:
> Added a new internal SentryOptions.ImplicitTransactionProcessors property

Why the new `Implicit` list? Can we add the `OpenTelemetryTransactionProcessor ` one we already have?

Reply:
Are we expecting the `Instrumenter` to be used by the user in any way? Right now it's for the SDK internally to know whether Sentry has been added to OpenTelemetry? I think this can be made `internal` then?

Reply:
> Why the new Implicit list? Can we add the OpenTelemetryTransactionProcessor one we already have?

I think my brain must have been frazzled when I was writing this. From memory, I saw `TransactionProcessors` has calls like `RemoveAll` on it, which would have complicated things... but looking again, I see that's only used to remove processors of specific types.

In any case, I'll get rid of the `ImplicitTransactionProcessors`

Reply:
> I think this can be made internal then?

Done 👍🏻